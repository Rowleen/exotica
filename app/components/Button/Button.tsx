'use client';
import { FC } from 'react';
import classNames from 'classnames';

import styles from './button.module.sass';

interface ButtonProps {
  color?: 'primary' | 'secondary';
  text: string;
  onClick: () => void;
  size?: 'small' | 'medium';
  type: 'pill' | 'button';
  value?: string;
}

const Button: FC<ButtonProps> = ({
  color,
  text,
  onClick,
  size,
  type,
  value
}) => {
  const buttonClass = classNames({
    [styles.button]: type === 'button',
    [styles.pill]: type === 'pill',
    [styles.primary]: color === 'primary',
    [styles.secondary]: color === 'secondary',
    [styles.medium]: size === 'medium',
    [styles.small]: size === 'small'
  });

  return (
    <button className={buttonClass} onClick={onClick} value={value}>
      {text}
    </button>
  );
};

export default Button;
