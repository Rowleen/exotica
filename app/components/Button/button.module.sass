@import '../../sass/variables'
@import '../../sass/fonts'

.button
  cursor: pointer
  border: solid 1px
  border-radius: $radius-md
  font-family: 'Roboto Regular'

.primary
  color: $font-color-secondary
  background-color: $bg-color-dark

  &:hover
    background-color: $font-color-third

.secondary
  color: $font-color-primary
  background-color: $bg-color-base

  &:hover
    background-color: $font-color-disabled

.medium
  font-size: 1rem
  line-height: 20px
  padding: $spacing-lg

.small
  font-size: .87rem
  line-height: 17.5px
  padding: $spacing-md

.pill
  cursor: pointer
  border: solid 1px $border-color-primary
  background-color: $bg-color-base
  padding: $spacing-lg $spacing-2xl $spacing-lg $spacing-3xl

  &:hover, &:active
    background-color: $bg-color-active

  &:first-child
    border-right: unset
    border-top-left-radius: $radius-rounded
    border-bottom-left-radius: $radius-rounded

  &:last-child
    border-left: unset
    border-top-right-radius: $radius-rounded
    border-bottom-right-radius: $radius-rounded
