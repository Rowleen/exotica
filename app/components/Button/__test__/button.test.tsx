import '@testing-library/jest-dom'
import { fireEvent, render } from '@testing-library/react'
import Button from '../Button'

describe('Button component', () => {
  it('renders button unchanged', () => {
    const { container } = render(
      <Button type='button' text='button' shape='button' />
    )
    expect(container).toMatchSnapshot()
  })

  it('fire onClick when user clicks', () => {
    const onClick = jest.fn()
    const { getByRole } = render(
      <Button type='button' text='button' shape='button' onClick={onClick} />
    )

    const button = getByRole('button', { name: 'button' })

    fireEvent.click(button)

    expect(onClick).toHaveBeenCalledTimes(1)
  })
})
