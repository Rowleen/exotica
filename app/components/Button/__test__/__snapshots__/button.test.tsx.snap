// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders button unchanged 1`] = `
<div>
  <button
    class="button"
    type="button"
  >
    button
  </button>
</div>
`;
